### ISSUE TITLE
Rendering fails if metadata does not exist

### ISSUE THREAD

**Issue by dereklieu** [jsvine/notebookjs#21]
First: very cool library!

I've found that [this line](https://github.com/jsvine/notebookjs/blob/master/notebook.js#L312) throws an exception when the `metadata` object does not exist. Since [metadata properties are all optional](https://nbformat.readthedocs.io/en/latest/format_description.html), I imagine there are other instances of notebooks in the wild that don't have this property set to an empty `{}`. 

Might be good to check that `metadata` is accessible to guard against throwing the exception in these cases.

Thanks!

**Comment by jsvine** [comment]
Thanks for flagging! Can you share a minimal notebook that raises this exception?

**Comment by dereklieu** [comment]
I've pasted such a file below. Using [your preview site](https://jsvine.github.io/nbpreview/), the file will not generate anything as-is. If I add `metadata: {}` to the root object then it generates the heading as it should.

I don't know if it's common for most notebook authoring programs not to include at least an empty `metadata` object, but I'm working with some here that don't so it's certainly possible!

```
{
  "cells": [
    {
      "level": 1,
      "source": [
        "Demo of mpld3 (Updated!)"
      ],
      "metadata": {},
      "cell_type": "heading"
    }
  ]
}
```

**Comment by jsvine** [comment]
Thanks! Fixed in v0.4.2, now available on `master` and on npm.
